### PR TITLE
Dump versions when we have a signal.

### PR DESCRIPTION
**Pull request by mmeeks**
Change-Id: I6b1f5ac1cebdd05dd9ff9541a5418059f55841ed
Signed-off-by: Michael Meeks <michael.meeks@collabora.com>

* Resolves: # 
* Target version: master 

### TODO

- [ ] ...

### Checklist

- [ ] Code is properly formatted
- [ ] All commits have Change-Id
- [ ] I have run tests with `make check`
- [ ] I have issued `make run` and manually verified that everything looks okay
- [ ] Documentation (manuals or wiki) has been updated or is not required

### CODE REVIEW

**Comment by mmeeks** [comment]
Would love a review - hopefully this helps with backtraces from unknown versions and lets us unwind symbols properly =)

File: common/SigUtil.cpp [hunk 1]
```diff
@@ -283,10 +289,19 @@ namespace SigUtil
         }
     }
 
-    void setFatalSignals()
+    void setVersionInfo(const std::string &versionInfo)
+    {
+        if (VersionInfo)
+            free (VersionInfo);
+        VersionInfo = strdup(versionInfo.c_str());
```
Comment (Ashod) [inline common/SigUtil.cpp:296]: I wonder if ASAN or some similar tool will complain about the loss of this allocation. I suppose we can whitelist it.

**Review by Ashod — APPROVED** [review]
Very useful stuff. Thanks!

File: common/SigUtil.cpp (hunk 1 above)
Comment (mmeeks) [reply common/SigUtil.cpp:296]: Possibly; but we can ignore that I think. Ultimately we could prolly use a std::string, and probably c_str() is signal safe - but, better to leak than to hide undefined behaviour in the code I think.

**Pull request merged**